src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/MacroParserImpl.java

```diff
@@ -61,44 +72,26 @@ public MacroParserImpl(TranslationUnit tu, Collection<String> args) {
      * If that is not possible (e.g. because the macro refers to other macro, or has a more complex grammar), fall
      * back to use clang evaluation support.
      */
-    Optional<Macro> eval(String macroName, String... tokens) {
-        if (tokens.length == 2) {
-            //check for fast path
-            Integer num = toNumber(tokens[1]);
-            if (num != null) {
-                return Optional.of(Macro.longMacro(Type.primitive(Type.Primitive.Kind.Int), num));
-            }
-        }
-        //slow path
-        try {
-            //step one, parse constant as is
-            MacroResult result = reparse(constantDecl(macroName, false));
-            if (!result.success()) {
-                //step two, attempt parsing pointer constant, by forcing it to uintptr_t
-                result = reparse(constantDecl(macroName, true)).asType(result.type);
-            }
-            return result.success() ?
-                    Optional.of((Macro)result) :
-                    Optional.empty();
-        } catch (BadMacroException ex) {
-            if (JextractTaskImpl.VERBOSE) {
-                System.err.println("Failed to handle macro " + macroName);
-                ex.printStackTrace(System.err);
+    @Override
+    public Optional<Declaration.Constant> parseConstant(Position pos, String name, String[] tokens) {
+        if (!(pos instanceof TreeMaker.CursorPosition)) {
+            return Optional.empty();
+        } else {
+            Cursor cursor = ((TreeMaker.CursorPosition)pos).cursor();
+            if (cursor.isMacroFunctionLike()) {
+                return Optional.empty();
+            } else if (tokens.length == 2) {
+                //check for fast path
+                Integer num = toNumber(tokens[1]);
+                if (num != null) {
+                    return Optional.of(treeMaker.createMacro(cursor, name, Type.primitive(Type.Primitive.Kind.Int), (long)num));
+                }
             }
+            macroTable.enterMacro(name, tokens, cursor);
             return Optional.empty();
```

Comment:
Better to have a DEBUG message for skipped macros. Useful for debugging. Perhaps try..catch RuntimeException with DEBUG/rethrow in catch is good as well.

Reply:
this is not skipped though - it is just deferred